src/components/SummaryConfirmation/index.js

```diff
@@ -1,55 +1,113 @@
 import {SUBMISSION_ALLOWED} from 'components/constants';
 
-export const PRIVACY_POLICY_ENDPOINT = 'config/privacy_policy_info';
+export const PRIVACY_POLICY_ENDPOINT = 'config/declarations_info_list';
 
-const getPrivacyPolicyInfo = async baseUrl => {
+const getDeclarationsInfo = async baseUrl => {
   return await get(`${baseUrl}${PRIVACY_POLICY_ENDPOINT}`);
 };
 
+const isSubmitEnabled = (loading, declarationsInfo = [], declarationsValues) => {
+  if (loading) return false;
+
+  return declarationsInfo.every(info => {
+    if (!info.validate.required) return true;
+
+    return Boolean(declarationsValues[info.key]);
+  });
+};
+
+const getDeclarationValues = (declarationsInfo = [], formikValues) => {
+  const declarationValuesAsArray = declarationsInfo
+    .map(info => {
+      if (!info.validate.required) return null;
+      return [info.key, formikValues[info.key] || false];
+    })
+    .filter(item => Array.isArray(item));
+
+  return Object.fromEntries(declarationValuesAsArray);
+};
+
+const getInitialWarningValues = declarationsValues => {
+  const warningValuesAsArray = Object.entries(declarationsValues).map(([declarationKey]) => [
+    declarationKey,
+    false,
+  ]);
+  return Object.fromEntries(warningValuesAsArray);
+};
+
 const SummaryConfirmation = ({submissionAllowed, onPrevPage}) => {
   const {baseUrl} = useContext(ConfigContext);
   const canSubmit = submissionAllowed === SUBMISSION_ALLOWED.yes;
 
   const {
     loading,
-    value: privacyPolicyConfig = {},
+    value: declarationsInfo = [],
     error,
   } = useAsync(async () => {
-    if (!canSubmit) return undefined;
-    return await getPrivacyPolicyInfo(baseUrl);
-  }, [baseUrl, getPrivacyPolicyInfo, canSubmit]);
+    if (!canSubmit) return [];
+
+    return await getDeclarationsInfo(baseUrl);
+  }, [baseUrl, getDeclarationsInfo, canSubmit]);
+  const {values: formikValues} = useFormikContext();
+
   if (error) throw error;
 
-  const {requiresPrivacyConsent, privacyLabel} = privacyPolicyConfig;
-  const {
-    values: {privacy: policyAccepted},
-  } = useFormikContext();
+  const submitDisabled = !isSubmitEnabled(loading, declarationsInfo, formikValues);
+
+  const [declarationsWarnings, setDeclarationWarnings] = useState(
+    getInitialWarningValues(getDeclarationValues(declarationsInfo, formikValues))
+  );
+
+  const showWarnings = formikValues => {
+    const declarationValues = getDeclarationValues(declarationsInfo, formikValues);
+
+    let updatedWarnings = {...declarationsWarnings};
+    Object.entries(declarationValues).forEach(([declarationKey, declarationValue]) => {
+      if (!declarationValue) {
+        updatedWarnings = {...updatedWarnings, [declarationKey]: true};
+      }
+    });
+    setDeclarationWarnings(updatedWarnings);
+  };
+
+  const declarationValues = getDeclarationValues(declarationsInfo, formikValues);
+  const previousDeclarationValues = usePrevious(declarationValues);
+
+  useEffect(() => {
```

Comment:
I've learned this myself recently, but you don't actually need `useEffect` to update state directly in this component, you can do it directly without the `useEffect` and that improves reactivity because you skip a render cycle (and it also makes the code a bit easier to follow).

This only works if you're calling the `setState` from `useState` hook in the same component, see also: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes